src/emerge.cpp

```diff
@@ -106,9 +106,9 @@ EmergeManager::EmergeManager(Server *server, MetricsBackend *mb)
 		m_qlimit_generate = nthreads + 1;
 
 	// don't trust user input for something very important like this
+	m_qlimit_total = rangelim(m_qlimit_total, 2, 1000000);
 	m_qlimit_diskonly = rangelim(m_qlimit_diskonly, 2, 1000000);
 	m_qlimit_generate = rangelim(m_qlimit_generate, 1, 1000000);
-	m_qlimit_total = std::max(m_qlimit_diskonly, m_qlimit_generate);
```

Comment:
oops. what I meant to do is to ensure the total limit isn't smaller than the other ones.
```c
	m_qlimit_total = std::max(m_qlimit_total, std::max(m_qlimit_diskonly, m_qlimit_generate));
```
or maybe this would make sense
```c
	m_qlimit_total = std::max(m_qlimit_total, m_qlimit_diskonly+ m_qlimit_generate);
```

Reply:
`m_qlimit_total = std::max(m_qlimit_total, std::max(m_qlimit_diskonly, m_qlimit_generate));`

Is good. `m_qlimit_diskonly + m_qlimit_generate` makes no sense as either one or the other is used (see also #15948)